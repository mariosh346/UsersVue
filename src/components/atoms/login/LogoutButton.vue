<template>
  <base-btn
    flat
    unelevated
    color="secondary"
    @click="handleSignout"
  >
    {{ $t('Logout') }}
  </base-btn>
</template>

<script setup lang="ts">
import { useUserStore } from 'src/stores/userStore';
import { useAuthFunctions } from 'src/composables/useAuthFunctions';
import BaseBtn from 'src/components/atoms/BaseBtn.vue';

const { signout } = useAuthFunctions();

const handleSignout = () => {

  signout()
    .then(() => {

      useUserStore().setUser(null);

    })
    .catch((error) => {

      console.error('Failed signout', error);

    });

};
</script>
