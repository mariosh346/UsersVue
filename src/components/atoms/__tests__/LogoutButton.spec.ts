import { describe, it, expect, beforeEach, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import LogoutButton from "../login/LogoutButton.vue"

// Replace jest.mock with vi.mock
vi.mock('../composables/useAuthFunctions', () => ({
  useAuthFunctions: () => ({
    signout: vi.fn(() => Promise.resolve())
  })
}))

describe('LogoutButton', () => {
  let wrapper: ReturnType<typeof shallowMount>
  const createWrapper = () => shallowMount(LogoutButton)

  beforeEach(() => {
    wrapper = createWrapper()
  })

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
