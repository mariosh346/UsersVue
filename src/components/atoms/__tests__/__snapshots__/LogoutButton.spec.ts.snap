// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogoutButton > matches snapshot 1`] = `"<base-btn-stub color="secondary" replace="false" outline="false" flat="true" unelevated="true" rounded="false" push="false" square="false" glossy="false" fab="false" fabmini="false" nocaps="false" nowrap="false" dense="false" ripple="false" stack="false" stretch="false" loading="false" disable="false" round="false" darkpercentage="false"></base-btn-stub>"`;
